src/pages/index.js

```diff
@@ -0,0 +1,312 @@
+  modal.addEventListener("click", (evt) => {
+    if (
+      evt.target.classList.contains("modal") ||
+      evt.target.classList.contains("modal__close-btn")
+    ) {
+      closeModal(modal);
+    }
+  });
+});
+
+// cancel delete confirmation modal
+
+const cancelCardModal = document.querySelector(
+  ".modal__button.modal__button_type_cancel"
+);
+cancelCardModal.addEventListener("click", function () {
+  closeModal(deleteCardModal);
+});
+// Delete modal logic
+const deleteCardModal = document.querySelector("#delete__card-modal");
+const deleteForm = deleteCardModal.querySelector(".modal__form");
+
+let selectedCard;
+let selectedCardId;
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteCardModal);
+}
+
+const handleDeleteSubmit = (evt) => {
+  evt.preventDefault();
+  const submitButton = deleteForm.querySelector(".modal__button_type_delete");
+  submitButton.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Yes"; // Reset button text
+    });
+};
+
+deleteForm.addEventListener("submit", handleDeleteSubmit);
+
+// Create card element
+function getCardElement(data, userId) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  cardElement.id = `card-${Date.now()}`;
+
+  const cardElementDescription =
+    cardElement.querySelector(".card__description");
+  const cardElementImage = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardElementDescription.textContent = data.name;
+  cardElementImage.src = data.link;
+  cardElementImage.alt = data.name;
+
+  cardLikeButton.addEventListener("click", () => {
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+    const likeAction = isLiked ? api.deleteLike : api.addLike;
+
+    likeAction
+      .call(api, data._id)
+      .then((updatedCard) => {
+        cardLikeButton.classList.toggle("card__like-button_liked");
+      })
+      .catch(console.error);
+  });
+
+  cardElementImage.addEventListener("click", () => {
+    previewImage.src = cardElementImage.src;
+    previewImage.alt = cardElementImage.alt;
+    previewCaption.textContent = cardElementImage.alt;
+    openModal(previewModal);
+  });
+
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  return cardElement;
+}
+
+// Handle profile form
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const profileEditModal = document.querySelector("#profile__edit-modal");
```

Comment:
`  const profileEditModal = document.querySelector("#profile__edit-modal");
`

You've already found this modal in line 41. Please, delete line 199